parlai/core/worlds.py

```diff
@@ -424,6 +428,8 @@ def __init__(self, opt, world):
             opti = copy.deepcopy(opt)
             opti['batchindex'] = i
             self.worlds.append(shared['world_class'](opti, None, shared))
+        self.batch_acts = [ [] ] * len(self.worlds)
```

Comment:
this needs to be: `self.batch_acts = [ [] for _ in self.worlds]`
because:
```python
>>> [ [] ] * 3
[[], [], []]
>>> l = [ [] ] * 3
>>> l[0].append(1)
>>> l
[[1], [1], [1]]
```